main.js

```diff
@@ -167,6 +167,11 @@ const fetchHandler = async function (req) {
   // Handle WebSocket connections
   if (urlPath[0] === "ws") {
 
+    // Load authentication header from parameters if present
+    // This is a workaround for internal requests, as the JavaScript WebSockets API doesn't support setting headers
+    const authSecret = url.searchParams.get("Authentication");
+    if (authSecret) req.headers.set("Authentication", authSecret);
+
```

Comment:
Putting secrets in URL parameters is generally strongly discouraged, as the URL may not be encrypted even if using TLS.

Quick read about the topic:
https://www.fullcontact.com/blog/2016/04/29/never-put-secrets-urls-query-parameters/
<sub>(I havent read this exact article in its entirety, but skimming over it, it seems to get the idea)</sub>